features/dual-stack-support.md

```diff
@@ -0,0 +1,229 @@
+[//]: # (The syntax preceeding this line is a comment marker used to help guide the author in populating this document)
+[//]: # (to github. Unlike HTML comments commonly used throughout istio.io documentation, this comment will not be rendered)
+[//]: # (by github. Comments must be separated by carriage return preceding and concluding the text and be a single line.)
+
+[//]: # (This is a living document representing the maturity of a feature. Completion of this template enables Istio work groups)
+[//]: # (to collect information on potential new functionality. This template should be completed before users are exposed to)
+[//]: # (any new experimental feature. Please complete this template during development.)
+
+[//]: # (The feature implementation section must be completed before submission of the document.)
+
+# Feature:
+
+[//]: # (All information in this section is mandatory.)
+
+**Feature name:**
+
+Dual Stack support for Kubernetes
+
+[//]: # (The name of the feature, e.g. Multiple control planes)
+
+**Primary lead(s):**
+
+Steve Zhang (huailong.zhang@intel.com), Jacob Delgado (j.delgado@f5.com)
+
+[//]: # (The primary lead or leads responsible for the feature. These individuals serve as a point of contact for the feature.)
+
+
+**Short description:**
+
+Add support for Dual Stack network model in Istio. With this, besides both IPv6 and IPv4 network model, Istio now supports dual-stack at the moment.
+
```

Comment:
Nit - is it Dual Stack/dual stack/ dual-stack? Let us try to use it consistently through the doc.

Reply:
It would also be nice to include somewhere a little description about this feature beyond just supporting dual stack in Istio or a user having to read the design doc to understand what is supported and what is not or what is upcoming?

We did work on that on the release notes for 1.17 to capture what is covered in 1.17 , so that could be what is supported in the alpha implementation.  Like others mentioned, we also need to include what docs/tests are there for dual stack. For docs I am assuming right now we only have that design doc. Maybe we can start thinking of including a doc under the traffic management section with an overview of this feature. From a tests perspective, do we have unit tests/integration tests - what all use cases does it cover, what needs to be added - those details may help reviewers. I do not mean to overwhelm you but just some suggestions.

Reply:
> Nit - is it Dual Stack/dual stack/ dual-stack? Let us try to use it consistently through the doc.

I suggest using Dual Stack because I always keep using this one, I can change if you have better advice.

Reply:
> It would also be nice to include somewhere a little description about this feature beyond just supporting dual stack in Istio or a user having to read the design doc to understand what is supported and what is not or what is upcoming?
> 
> We did work on that on the release notes for 1.17 to capture what is covered in 1.17 , so that could be what is supported in the alpha implementation. Like others mentioned, we also need to include what docs/tests are there for dual stack. For docs I am assuming right now we only have that design doc. Maybe we can start thinking of including a doc under the traffic management section with an overview of this feature. From a tests perspective, do we have unit tests/integration tests - what all use cases does it cover, what needs to be added - those details may help reviewers. I do not mean to overwhelm you but just some suggestions.

Hi @johnma14, thanks for your suggestions, as mentioned in blog pr: https://github.com/istio/istio.io/pull/12253 by @jacob-delgado, there would be other docs for dual stack support, including on how to enable dual stack feature and what's the use case for dual stack? And we can work on that together. ^-^

Reply:
Sounds good @zhlsunshine.